internal/driver/controllerserver_helper_test.go

```diff
@@ -477,26 +467,31 @@ func TestPrepareVolumeParams(t *testing.T) {
 			}
 			ctx := context.Background()
 
-			volumeName, sizeGB, size, _, err := cs.prepareVolumeParams(ctx, tt.req)
+			params, err := cs.prepareVolumeParams(ctx, tt.req)
 
-			if err != nil && !reflect.DeepEqual(tt.expectedError, err) {
-				if tt.expectedError != nil {
-					t.Errorf("expected error %v, got %v", tt.expectedError, err)
-				} else {
-					t.Errorf("expected no error but got %v", err)
-				}
+			// First, verify that the error matches the expectation
+			if (err != nil && tt.expectedError == nil) || (err == nil && tt.expectedError != nil) {
+				t.Errorf("expected error %v, got %v", tt.expectedError, err)
+			} else if err != nil && tt.expectedError != nil && err.Error() != tt.expectedError.Error() {
+				t.Errorf("expected error message %v, got %v", tt.expectedError.Error(), err.Error())
 			}
 
-			if !reflect.DeepEqual(volumeName, tt.expectedName) {
-				t.Errorf("Expected volume name: %s, but got: %s", tt.expectedName, volumeName)
-			}
+			// Only check params fields if params is not nil
+			if params != nil {
```

Comment:
question: Is there a case where params will be nil?

Reply:
Yup, yesterday i ran into a null pointer issue when the volume sizes were set to negative numbers. So i added this check.